modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/SkuResourceImpl.java

```diff
@@ -245,9 +250,22 @@ public Sku postProductIdSku(Long id, Sku sku) throws Exception {
 	private Sku _addOrUpdateSKU(CPDefinition cpDefinition, Sku sku)
 		throws Exception {
 
+		ServiceContext serviceContext = _serviceContextHelper.getServiceContext(
+			cpDefinition.getGroupId());
+
 		CPInstance cpInstance = SkuUtil.addOrUpdateCPInstance(
-			_cpInstanceService, sku, cpDefinition,
-			_serviceContextHelper.getServiceContext(cpDefinition.getGroupId()));
+			_cpInstanceService, sku, cpDefinition, serviceContext);
+
+		if (Objects.equals(
+				SkuUtil.getCommercePricingConfigurationKey(
+					_configurationProvider),
+				CommercePricingConstants.VERSION_2_0)) {
+
+			SkuUtil.updateCommercePriceEntries(
+				_commercePriceEntryLocalService, _commercePriceListLocalService,
+				cpInstance, cpInstance.getPrice(), cpInstance.getPromoPrice(),
```

Comment:
This is not fixing the bug. The bug aims to add the possibility to save Cast, Price and PromoPrice via API just link this 
`// Update pricing info

		BigDecimal price = (BigDecimal)ParamUtil.getNumber(
			actionRequest, "price", BigDecimal.ZERO);
		BigDecimal promoPrice = (BigDecimal)ParamUtil.getNumber(
			actionRequest, "promoPrice", BigDecimal.ZERO);
		BigDecimal cost = (BigDecimal)ParamUtil.getNumber(
			actionRequest, "cost", BigDecimal.ZERO);

		cpInstance = _cpInstanceService.updatePricingInfo(
			cpInstance.getCPInstanceId(), price, promoPrice, cost,
			serviceContext);

		if (Objects.equals(
				_getCommercePricingConfigurationKey(),
				CommercePricingConstants.VERSION_2_0)) {

			_updateCommercePriceEntries(
				cpInstance, price, promoPrice,
				ServiceContextFactory.getInstance(actionRequest));
		}
`

From the class EditCPInstanceMVCActionCommand

Reply:
So you need to fetch those fields from the Sku object not from the CPInstance

Reply:
@andrea-ale-sbarra , correct me if I am wrong, but the call is being made, after the SKU object is saved as a CPInstance object. Therefore, I am saving the fields from the SKU object.

Reply:
Yes I meant to make the same logic not to actually call updatePricingInfo method. But still you are not making sure that the values that you pass via REST call are actually saved in this implementation.. What if another call is made in another parallel thread with different values?

Reply:
Also I tried your fix locally and this is not working when I'm trying to PATCH the SKU